app/scripts/lib/tx-utils.js

```diff
@@ -107,20 +83,17 @@ module.exports = class txProviderUtils {
   }
 
   publishTransaction (rawTx) {
-    return new Promise((resolve, reject) => {
-      this.query.sendRawTransaction(rawTx, (err, ress) => {
-        if (err) reject(err)
-        else resolve(ress)
-      })
-    })
+    return this.query.sendRawTransaction(rawTx)
```

Comment:
this is correct but lets use the `async` keyword to make it clear we're returning a promise
```js
async publishTransaction (rawTx) {
  return await this.query.sendRawTransaction(rawTx)
}
```